### ISSUE TITLE
Stupid quesiton about loss function 'loss = F.mse_loss(latents, (latents - grad).detach())'

### ISSUE THREAD

**Issue by xiangyuxu0815** [cswry/OSEDiff#54]
In the code here:

https://github.com/cswry/OSEDiff/blob/fcc4f09c7ed66a9bc0ba9433940d76b343142426/osediff.py#L241

Are 'loss = F.mse_loss(latents, (latents - grad).detach())' equal to 'loss = F.mse_loss(grad.detach(), torch.zeros_like(grad))'? Why have to include 'latents' here? Is there any specific purpose?
 
Best,

**Comment by CuddleSabe** [comment]
> In the code here:
> 
> https://github.com/cswry/OSEDiff/blob/fcc4f09c7ed66a9bc0ba9433940d76b343142426/osediff.py#L241
> 
> Are 'loss = F.mse_loss(latents, (latents - grad).detach())' equal to 'loss = F.mse_loss(grad.detach(), torch.zeros_like(grad))'? Why have to include 'latents' here? Is there any specific purpose?
> 
> Best,

不然mse的loss会通过target来将梯度传给latents

**Comment by xiangyuxu0815** [comment]
> > In the code here:
> > https://github.com/cswry/OSEDiff/blob/fcc4f09c7ed66a9bc0ba9433940d76b343142426/osediff.py#L241
> > 
> > Are 'loss = F.mse_loss(latents, (latents - grad).detach())' equal to 'loss = F.mse_loss(grad.detach(), torch.zeros_like(grad))'? Why have to include 'latents' here? Is there any specific purpose?
> > Best,
> 
> 不然mse的loss会通过target来将梯度传给latents
感谢！
